src/main/java/org/zaproxy/zap/extension/jwt/JWTActiveScanRule.java

```diff
@@ -71,6 +79,29 @@ public void init() {
         }
     }
 
+    protected void scan(List<NameValuePair> nameValuePairs) {
+        if (nameValuePairs.isEmpty()
+                || nameValuePairs.get(0).getType() != NameValuePair.TYPE_HEADER) {
+            super.scan(nameValuePairs);
+        } else {
```

Comment:
`if (!nameValuePairs.isEmpty() && nameValuePairs.get(0).getType() == NameValuePair.TYPE_HEADER)) {` it can then call `super.scan(nameValuePairs);` just once.

Reply:
but what will happen when this method is called for other variants?

Reply:
ignore, got your point.

Reply:
This does not change the existing logic, for other variants the type is different and will just call `super.scan(nameValuePairs);`.